src/main/java/in/org/projecteka/hiu/consent/ConsentService.java

```diff
@@ -71,16 +77,33 @@
     }
 
     private Flux<Void> upsertConsentArtefacts(ConsentNotificationRequest consentNotificationRequest) {
-        return Flux.fromIterable(consentNotificationRequest.getConsents())
-                .flatMap(consentArtefactReference ->
-                        consentArtefactReference.getStatus() == ConsentStatus.GRANTED
-                        ? processGrantedConsent(consentArtefactReference,
-                                consentNotificationRequest.getConsentRequestId())
-                        : processRejectedConsent(consentArtefactReference));
+        return Flux.fromIterable(consentNotificationRequest.getConsentArtefacts())
+                .flatMap(consentArtefactReference -> {
+                    if(consentNotificationRequest.getStatus() == ConsentStatus.GRANTED)
+                        return processGrantedConsent(consentArtefactReference,
+                                consentNotificationRequest.getConsentRequestId());
+                    else if(consentNotificationRequest.getStatus() == ConsentStatus.REVOKED)
+                        return processRevokedConsent(consentArtefactReference,
+                                consentNotificationRequest.getStatus(),
+                                consentNotificationRequest.getTimestamp());
+                    else
+                        return processRejectedConsent(consentArtefactReference,
+                                consentNotificationRequest.getStatus(),
+                                consentNotificationRequest.getTimestamp());
+                });
+    }
+
+    private Mono<Void> processRevokedConsent(ConsentArtefactReference consentArtefactReference,
+                                             ConsentStatus status,
+                                             Date timestamp) {
+        return consentRepository.updateStatus(consentArtefactReference, status, timestamp)
+                .then(healthInfoDeletionPublisher.broadcastHealthInfoDeletionRequest(consentArtefactReference));
```

Comment:
does it make sense to publish the revoke message into the same queue where we publish consent-approval message? and consumer decide what to do based on status?

or can we rename the class and queue to something else instead of literally saying 'HealthInfoDeletionPublisher'.